src/components/buttons/Button.spec.tsx

```diff
@@ -1,87 +1,306 @@
 import * as React from 'react';
-import {Spinner, Icon} from '../..';
+import {Icon} from '../..';
 import Button, {BUTTON_VARIANT} from './Button';
-import {shallow} from 'enzyme';
+import {render, within} from '@testing-library/react';
+import {testA11y} from '../../axe';
+import userEvent from '@testing-library/user-event';
 
-test('render', () => {
-  const button = shallow(<Button>Some text</Button>);
+describe('Button', () => {
+  test('render', () => {
+    const button = render(<Button>Some text</Button>);
+    const root = button.container.firstElementChild;
 
-  expect(button.hasClass('sg-button')).toEqual(true);
-});
-test('variant', () => {
-  const buttonVariant = BUTTON_VARIANT.SOLID;
-  const button = shallow(<Button variant={buttonVariant}>Some text</Button>);
+    expect(root.classList.contains('sg-button')).toEqual(true);
+  });
 
-  expect(button.hasClass(`sg-button--${buttonVariant}`)).toEqual(true);
-});
-test('disabled', () => {
-  const button = shallow(<Button disabled>Some text</Button>);
+  test('variant', () => {
+    const buttonVariant = BUTTON_VARIANT.SOLID;
+    const button = render(<Button variant={buttonVariant}>Some text</Button>);
+    const root = button.container.firstElementChild;
 
-  expect(button.hasClass('sg-button--disabled')).toEqual(true);
-  expect(button.is('[disabled]')).toEqual(true);
-});
-test('not disabled', () => {
-  const button = shallow(<Button>Some text</Button>);
+    expect(root.classList.contains(`sg-button--${buttonVariant}`)).toEqual(
+      true
+    );
+  });
 
-  expect(button.hasClass('sg-button--disabled')).toEqual(false);
-  expect(button.is('[disabled]')).toEqual(false);
-});
-test('full width', () => {
-  const button = shallow(<Button fullWidth>Some text</Button>);
+  test('disabled', () => {
+    const button = render(<Button disabled>Some text</Button>);
+    const root = button.container.firstElementChild;
 
-  expect(button.hasClass('sg-button--full-width')).toEqual(true);
-});
-test('icon', () => {
-  const icon = <span>:P</span>;
-  const button = shallow(<Button icon={icon}>Some text</Button>);
+    expect(root.classList.contains('sg-button--disabled')).toEqual(true);
+    expect(root.hasAttribute('disabled')).toEqual(true);
```

Comment:
I think `expect(button.getByRole('button')).toHaveProperty('disabled', true)` would be better - it tests actually if the button is disabled and does not rely on structure.